packages/jellyfish-transaction/src/script/dftx/dftx_masternode.ts

```diff
@@ -1,30 +1,42 @@
+import { SmartBuffer } from 'smart-buffer'
 import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
 
 // Disabling no-return-assign makes the code cleaner with the setter and getter */
 /* eslint-disable no-return-assign */
 
 /**
- * CreateMasterNode DeFi Transaction
+ * CreateMasternode DeFi Transaction
  */
-export interface CreateMasterNode {
-  type: number // ----------------------------------| 1 byte, if 0x01 p2pkh, otherwise = p2wpkh
-  collateralPubKeyHash: string // ------------------| VarUInt{20 bytes}
-  operatorPubKeyHash?: string // -------------------| VarUInt{20 bytes}, p2pkh only, optional, use collateral pkh as default
+export interface CreateMasternode {
+  operatorType: number // --------------------------| 1 byte, 0x01 = p2pkh, 0x04 = p2wpkh
+  operatorAuthAddress: string // -------------------| VarUInt{20 bytes}
+  timelock?: number // -----------------------------| 2 bytes, 0x0401 = 5 years, 0x0802 = 10 years
 }
 
 /**
- * Composable CreateMasterNode, C stands for Composable.
+ * Composable CreateMasternode, C stands for Composable.
  * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
  */
-export class CCreateMasterNode extends ComposableBuffer<CreateMasterNode> {
+export class CCreateMasternode extends ComposableBuffer<CreateMasternode> {
   static OP_CODE = 0x43 // 'C'
   static OP_NAME = 'OP_DEFI_TX_CREATE_MASTER_NODE'
 
-  composers (cmn: CreateMasterNode): BufferComposer[] {
+  composers (cmn: CreateMasternode): BufferComposer[] {
     return [
-      ComposableBuffer.uInt8(() => cmn.type, v => cmn.type = v),
-      ComposableBuffer.hex(20, () => cmn.collateralPubKeyHash, v => cmn.collateralPubKeyHash = v),
-      ComposableBuffer.optionalHex(20, () => cmn.operatorPubKeyHash, v => cmn.operatorPubKeyHash = v)
+      ComposableBuffer.uInt8(() => cmn.operatorType, v => cmn.operatorType = v),
+      ComposableBuffer.hex(20, () => cmn.operatorAuthAddress, v => cmn.operatorAuthAddress = v),
+      {
+        fromBuffer: (buffer: SmartBuffer): void => {
+          if (buffer.remaining() > 0) {
+            cmn.timelock = buffer.readUInt16LE()
+          }
+        },
+        toBuffer: (buffer: SmartBuffer): void => {
+          if (cmn.timelock !== undefined) {
```

Comment:
After eunosPaya, timelock value(0x0000) will be added when serializing if not specified. this will change the total length to 28(0x1c). nodes might still parse it to default value 0 even if we don't send it. But better to implement accordingly with defid. 
so have a default value 0 for timelock and write that when serializing.
This is what is get for scriptPubkey without timelock and with it.
```
6a 1c 44665478 43 04 c93c4da5a164f96c18f5abc1c7534bd2d7036199 0000
6a 1c 44665478 43 04 dd8537bb5d1f069f8d4c66770a9fc705373ba98c 0401
```